### PR TITLE
feat(Dialog): don't override type attribute if set on mat-dialog-close

### PR DESCRIPTION
**Pull request by flauc**
only default the type attribute on mat-dialog-close if it isn't set

#16909

### CODE REVIEW

**Comment by jelbourn** [comment]
Can you update the API gold? Just run 
```
yarn bazel run //tools/public_api_guard:material/dialog.d.ts_api.accept
```

(we have golds for d.ts files to prevent public APIs from being changed inadvertently)

File: src/material/dialog/dialog-content-directives.ts [hunk 1]
```diff
@@ -30,13 +30,16 @@ let dialogElementUid = 0;
...
-    'type': 'button', // Prevents accidental form submits.
+    '[attr.type]': 'buttonType',
   }
 })
 export class MatDialogClose implements OnInit, OnChanges {
   /** Screenreader label for the button. */
   @Input('aria-label') ariaLabel: string;
 
+  /** Default to "button" to prevents accidental form submits. */
+  @Input('type') buttonType: 'submit' | 'button' = 'button';
```
Comment (crisbeto) [inline src/material/dialog/dialog-content-directives.ts]: Buttons can also have the `reset` type.

File: src/material/dialog/dialog-content-directives.ts [hunk 2]
```diff
@@ -30,13 +30,16 @@ let dialogElementUid = 0;
   host: {
     '(click)': 'dialogRef.close(dialogResult)',
     '[attr.aria-label]': 'ariaLabel || null',
-    'type': 'button', // Prevents accidental form submits.
+    '[attr.type]': 'buttonType',
```
Comment (crisbeto) [inline src/material/dialog/dialog-content-directives.ts]: I think that we can call the property just `type` since the element itself is a button.

**Comment by flauc** [comment]
> yarn bazel run //tools/public_api_guard:material/dialog.d.ts_api.accept

I ran it and it completed successfully but I didn't see anything change in the source?

**Comment by flauc** [comment]
The failing tests don't look to be related to the dialog.

**Pull request merged**